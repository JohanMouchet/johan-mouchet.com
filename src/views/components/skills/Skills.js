// @flow

import * as React from "react";
import { ProgressBar } from "views/objects";

type Props = {
  skills: typeof ProgressBar,
};

const Skills = ({ skills }: Props) =>
  !skills ? null : (
    <ul className="c-skills o-list--unstyled">
      {skills.map((skill) => (
        <li className="c-skill" key={skill.label}>
          <ProgressBar
            label={skill.label}
            filling={skill.filling}
            tooltip={skill.tooltip}
            tooltipOnHover
          />
        </li>
      ))}
    </ul>
  );

export default Skills;
