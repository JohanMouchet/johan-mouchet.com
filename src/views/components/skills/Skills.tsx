import React from "react";
import { ProgressBar } from "views/objects";
import styles from "views/objects/list/List.module.scss";

type Props = {
  skills: Pick<
    React.ComponentProps<typeof ProgressBar>,
    "label" | "filling" | "progress"
  >[];
};

const Skills: React.FC<Props> = ({ skills }) =>
  !skills?.length ? null : (
    <ul className={styles["o-list--unstyled"]}>
      {skills.map((skill) => (
        <li key={skill.label}>
          <ProgressBar
            label={skill.label}
            filling={skill.filling}
            progress={skill.progress}
            total="10"
            detailsOnHover
          />
        </li>
      ))}
    </ul>
  );

export default Skills;
