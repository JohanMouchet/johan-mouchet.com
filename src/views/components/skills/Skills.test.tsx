import React from "react";
import { render } from "@testing-library/react";
import Skills from ".";

it("Renders <Experience /> without crashing", () => {
  expect(
    render(
      <Skills
        skills={[
          {
            label: "Label",
            filling: 50,
            progress: "Progress",
          },
        ]}
      />
    ).container.firstChild
  ).toMatchSnapshot();
});
