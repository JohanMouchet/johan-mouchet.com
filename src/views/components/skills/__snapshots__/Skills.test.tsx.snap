// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Experience /> without crashing 1`] = `
<ul
  class="o-list--unstyled"
>
  <li>
    <div
      class="o-progress-bar o-progress-bar--details-on-hover"
      tabindex="0"
    >
      <span
        class="o-progress-bar__labels"
      >
        <span
          class="o-progress-bar__label"
        >
          Label
        </span>
        <span
          class="o-progress-bar__details"
        >
          <span
            class="o-progress-bar__progress"
          >
            Progress
          </span>
           / 
          <span
            class="o-progress-bar__total"
          >
            10
          </span>
        </span>
      </span>
      <div
        class="o-progress-bar__container"
      >
        <div
          class="o-progress-bar__filling"
          style="width: 50%;"
        />
      </div>
    </div>
  </li>
</ul>
`;
